### PR TITLE
Refactor Mbed erase

### PR DESCRIPTION
**Pull request by juhhov**
## Status
**READY**

## Description
Remove duplication between mbed board flash, erase and reset. Make erase more robust.

## Todos
- [x] Tests

### CODE REVIEW

File: mbed_flasher/common.py [hunk 1]
```diff
@@ -19,6 +19,10 @@
 import six
 
 
+from mbed_flasher.return_codes import EXIT_CODE_COULD_NOT_MAP_DEVICE
+from mbed_flasher.return_codes import EXIT_CODE_UNHANDLED_EXCEPTION
```
Comment (jonikula) [inline mbed_flasher/common.py]: Don't these imports fit on one line?

File: mbed_flasher/common.py [hunk 2]
```diff
@@ -50,34 +53,38 @@ def get_devices():
...
-            return list({dev["target_id"]: dev for dev in available_devices}.values())
+            found = list({dev["target_id"]: dev for dev in available_devices}.values())
+
+            # If not searching for specific target, just return what is found.
+            if not required_target_id:
+                return found
+
+            try:
+                for target_id in [device["target_id"] for device in found]:
+                    if target in target_id:
```
Comment (YifeiZuo01) [inline mbed_flasher/common.py]: is this looking for required_target_id when it's not None?

File: mbed_flasher/flashers/FlasherMbed.py [hunk 3]
```diff
@@ -313,8 +198,7 @@ def copy_file(self, source, destination):
         self.logger.debug("SHA1: %s", hashlib.sha1(aux_source).hexdigest())
 
         if platform.system() == 'Windows':
-            self.logger.debug("copying file: \"%s\" to \"%s\"",
-                              source, destination)
+            self.logger.debug('copying file: "{}" to "{}"', source, destination)
```
Comment (YifeiZuo01) [inline mbed_flasher/flashers/FlasherMbed.py]: logging here seems not working as expected.

File: mbed_flasher/erase.py [hunk 4]
```diff
@@ -67,90 +63,44 @@ def __get_flashers():
...
-            if err:
-                self.logger.debug("Remount due to erase")
-                break
-
-            if time() - start_time > ERASE_REMOUNT_TIMEOUT:
-                self.logger.debug("Didn't notice device to disappear for remount")
-                break
+            with open(join(target["mount_point"], 'DETAILS.TXT'), 'rb') as new_file:
+                details_txt = new_file.readlines()
+        except IOError:
```
Comment (YifeiZuo01) [inline mbed_flasher/erase.py]: For python 3, here will raise OSError for open, [doc](https://docs.python.org/3/library/functions.html#open)

File: mbed_flasher/common.py (hunk 1 above)
Comment (juhhov) [reply mbed_flasher/common.py]: Barely.. :)

**Comment by juhhov** [comment]
systest@raasdesktop:~/workspace/mbed-flasher$ coverage run -m unittest discover -s test
.usage: mbedflash [-h] [-v] [-s] <command> ...
mbedflash: error: argument <command>: invalid choice: 'discover' (choose from 'list', 'flashers', 'version', 'flash', 'reset', 'erase')
......[ERROR](mbed-flasher): Flashing failed: The transfer timed out.. tid=0240000034544e45004b00018aa9001d2011000097969900
[ERROR](mbed-flasher): Failed: The transfer timed out.
....................usage: mbedflash [-h] [-v] [-s] <command> ...
mbedflash: error: too few arguments
...usage: mbedflash [-h] [-v] [-s] <command> ...
mbedflash: error: argument <command>: invalid choice: 'discover' (choose from 'list', 'flashers', 'version', 'flash', 'reset', 'erase')
..................usage: mbedflash [-h] [-v] [-s] <command> ...

For specific command help, run: mbedflash <command> --help

optional arguments:
  -h, --help     show this help message and exit
  -v, --verbose  Verbose level... repeat up to three times.
  -s, --silent   Silent - only errors will be printed.

command:
  <command>      command help
    list         Prints a list of supported platforms.
    flashers     Prints a list of supported flashers.
    version      Display version information
    flash        Flash given resource
    reset        Reset given resource
    erase        Erase given resource
.usage: mbedflash [-h] [-v] [-s] <command> ...
mbedflash: error: argument <command>: invalid choice: 'a' (choose from 'list', 'flashers', 'version', 'flash', 'reset', 'erase')
..s.s...............................................................
----------------------------------------------------------------------
Ran 117 tests in 433.871s

OK (skipped=2)

**Pull request merged**